### ISSUE TITLE
0.13 regression in `-X watch`: `-x` Command parsing broken

### ISSUE THREAD

**Issue by LudvigHz** [tectonic-typesetting/tectonic#1040]
Just updated to 0.13.0. In this release, the `watch` command seems to not properly parse arguments properly (or argument parsing in general seems a bit broken?)

E.g. `tectonic -X watch -x compile main.tex`. The `main.tex` argument is passed to the `-X watch` command, giving an error.:
```
$ tectonic -X watch -x compile main.tex
note: "version 2" Tectonic command-line interface activated
[Running `tectonic -X compile`]
error: The following required arguments were not provided:
    <input>

USAGE:
    compile <input> --outfmt <format> --pass <pass> --format <path>

For more information try --help
[Running `tectonic -X main.tex`]
error: Found argument 'main.tex' which wasn't expected, or isn't valid in this context

USAGE:
    tectonic -X [OPTIONS] <SUBCOMMAND>

```

Quoting the command to run does not split the arguments either (This is what used to work correctly in `0.12.0`): `tectonic -X watch -x "compile main.tex"`:
```
$ tectonic -X watch -x "compile main.tex"
note: "version 2" Tectonic command-line interface activated
[Running `tectonic -X compile main.tex`]
error: The subcommand 'compile main.tex' wasn't recognized
        Did you mean 'compile'?

If you believe you received this message in error, try re-running with 'tectonic -X -- compile main.tex'

USAGE:
    tectonic -X [OPTIONS] <SUBCOMMAND>

```

The suggested command does not work either, giving this really quite funny error message:
```
$ tectonic -X -- compile main.tex --synctex
error: The subcommand 'compile' wasn't recognized
        Did you mean 'compile'?

If you believe you received this message in error, try re-running with 'tectonic -X -- compile'

USAGE:
    tectonic -X [OPTIONS] <SUBCOMMAND>

```

System information:
```
OS: Arch Linux (Updated 21.05.2023)
Kernel: 6.3.2-arch1-1
```

----

Downgrading to `0.12.0` as the command works as expected here (`tectonic -X watch -x "compile main.tex --synctex"`)

**Comment by CraftSpider** [comment]
I suspect this is related to changes from the `watchexec` update - I'll test it out locally and PR a fix to the handling

**Comment by pkgw** [comment]
This should be fixed in the 0.13.1 release that went out yesterday. Thanks for reporting!
